Update documentation on package resources (#9211)
ballerina-platform/ballerina-dev-website: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md

Writing code in an organized manner from the beginning of the project is important for the lifecycle of the project and its maintainability in the long run. Organized code will make it easy to extend and improve your project over time. Ballerina project structure makes it easy to write clean code by eliminating repetitions, writing reusable code, adding new features without changing the existing code, etc. To achieve this, Ballerina has the concept of packages and modules. 

Ballerina code is organized in a single shareable unit called a package.

> A package is a collection of modules, and a module is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can contain resources and test-only resources specific to the package in the package source root.

Reviewer: How is a resource defined as test-only?

Author: when the resources are added to a `resources` directory within `tests` directory.

Author: ```suggestion
A package is a collection of modules, and a module is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can include resources in a `resources` directory located at the package source root, and test-specific resources in a `resources` directory within the `tests` directory at the package source root.
```

Reviewer: If we have test-specific resources for submodules, they also have to go in the tests directory in the root?

Author: we don't support resources for sub modules. Resources are applicable to the entire package not module specific.

> Ballerina code is organized in a single shareable unit called a `package`.

Reviewer: Not introduced by this PR, but shall we remove backticks around package, module, etc. We use backticks only for keywords, references to names, etc.

Author: Addressed with 3e84be21c2dc33076f5c69ca3fe0b9bba41a2b89

> A `package` is a collection of `modules`, and a `module` is a collection of Ballerina source files, and test files. A package should contain at least one module called the default module. Each module has its own directory, which organizes source files, and test files. A package can contain resources and test-only resources specific to the package in the package source root.

Reviewer: ```suggestion
A package is a collection of modules, and a `module` is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can contain resources and test-only resources specific to the package in the package source root.
```
